src/index/hgraph_index.cpp

```diff
@@ -209,7 +211,7 @@ HGraphIndex::hnsw_add(const DatasetPtr& data) {
     auto* ids = data->GetIds();
     auto* datas = data->GetFloat32Vectors();
     auto cur_count = this->bottom_graph_->TotalCount();
-    vsag::Vector<std::shared_mutex>(total + cur_count, allocator_).swap(this->neighbors_mutex_);
+    this->resize(total + cur_count);
```

Comment:
is it ok if we call hnsw_add frequently

Reply:
we need a strategy for resize, less than 1000 increase 100, more than 1000 increase 1000? like this